source/MaterialXView/ShaderValidators/Glsl/GlslValidator.cpp

```diff
@@ -0,0 +1,1225 @@
+    {
+        return _stages[stage];
+    }
+    return std::string();
+}
+
+void GlslValidator::clearStages()
+{
+    for (size_t i = 0; i < HwShader::NUM_STAGES; i++)
+    {
+        _stages[i].clear();
+    }
+
+    // Clearing stages invalidates any cached inputs
+    clearInputLists();
+}
+
+bool GlslValidator::haveValidStages() const
+{
+    // Need at least a pixel shader stage and a vertex shader stage
+    const std::string& vertexShaderSource = _stages[HwShader::VERTEX_STAGE];
+    const std::string& fragmentShaderSource = _stages[HwShader::PIXEL_STAGE];
+
+    return (vertexShaderSource.length() > 0 && fragmentShaderSource.length() > 0);
+}
+
+
+void GlslValidator::initialize()
+{
+    ShaderValidationErrorList errors;
+    const std::string errorType("OpenGL utilities initialization.");
+
+    if (!_initialized)
+    {
+        // Creeate window
+        SimpleWindow window;
+        const char* windowName = "Validator Window";
+        bool created = window.create(const_cast<char *>(windowName),
+                                    _frameBufferWidth, _frameBufferHeight,
+                                    nullptr);
+        if (!created)
+        {
+            errors.push_back("Failed to create window for testing.");
+            throw ExceptionShaderValidationError(errorType, errors);
+        }
+        else
+        {
+            // Create offscreen context
+            GLUtilityContext* context = GLUtilityContext::create(window.windowWrapper(), nullptr);
+            if (!context)
+            {
+                errors.push_back("Failed to create OpenGL context for testing.");
+                throw ExceptionShaderValidationError(errorType, errors);
+            }
+            else
+            {
+                if (context->makeCurrent())
+                {
+                    // Initialize glew
+                    bool initializedFunctions = true;
+
+                    glewInit();
+                    if (!glewIsSupported("GL_VERSION_4_0"))
+                    {
+                        initializedFunctions = false;
+                        errors.push_back("OpenGL version 4.0 not supported");
+                    }
+
+                    if (initializedFunctions)
+                    {
+                        glClearColor(0.8f, 0.8f, 0.8f, 1.0f);
+                        glClearStencil(0);
+
+                        _initialized = true;
+                    }
+                }
+            }
+        }
+    }
+}
+
+void GlslValidator::deleteTarget()
+{
+    if (_frameBuffer)
+    {
+        GLUtilityContext* context = GLUtilityContext::get();
+        if (context && context->makeCurrent())
+        {
+            glBindFramebuffer(GL_FRAMEBUFFER, UNDEFINED_OPENGL_RESOURCE_ID);
+            glDeleteTextures(1, &_colorTarget);
+            glDeleteTextures(1, &_depthTarget);
+            glDeleteFramebuffers(1, &_frameBuffer);
+        }
+    }
+}
+
+bool GlslValidator::createTarget()
```

Comment:
Offscreen target code. Just hard-coded int FP color and depth. FP color we need for color management ops to work without degradation. I'm following what is used for GPU testing of OCIO.